### PR TITLE
Improve syntax highlighting

### PR DESCRIPTION
**Pull request by KilDesu**
## PR Type

> What kind of change does this PR introduce?

- [ ] Bugfix
- [x] Feature
- [ ] Documentation
- [ ] Code style update
- [x] Refactor
- [ ] Build-related changes
- [ ] Other, please describe:

## Does this PR introduce a breaking change?

- [ ] Yes
- [x] No

## What's new?

> List the changes

- Give "add-transparency" mixin a better name
- Replace highlightjs and svelte-highlight with shiki, allowing more languages and themes

## Screenshots

> If needed, you can add screenshots here

N/A

## This pull request closes an issue

> Add `Closes`, `Fixes` or `Resolves` followed by `#xxx[,#xxx]` where "xxx" is the issue number

N/A

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -70,9 +72,5 @@
   },
   "svelte": "./dist/index.js",
   "types": "./dist/index.d.ts",
-  "type": "module",
-  "dependencies": {
-    "highlight.js": "^11.10.0",
```
Comment (therealcoder1337) [inline package.json:75]: It's good to remove them, but I'm a bit surprised we don't have anything in `dependencies` now. We should check this once we deploy the docs.

File: src/lib/components/private/QDocsSection.svelte [hunk 2]
```diff
@@ -13,15 +13,17 @@
...
 
   let dialog = $state(false);
+
+  const code = $derived(snippets?.[title].replaceAll(/^  /gm, ""));
 </script>
 
 <div style="margin-bottom:48px">
   <div class="flex justify-between q-mb-md">
     <h5>{title}</h5>
-    {#if snippets}
```
Comment (therealcoder1337) [inline src/lib/components/private/QDocsSection.svelte:21]: Shouldn't `snippets` be removed from all the code here?

File: src/lib/css/mixins/_design.scss [hunk 3]
```diff
@@ -7,8 +7,7 @@
   }
   $elevate: 0.125rem * $strength;
 
-  box-shadow: 0 $elevate * $coeff $elevate 0
-    add-transparency(var(--shadow), calc(24% + 8% * $strength));
+  box-shadow: 0 $elevate * $coeff $elevate 0 with-alpha(var(--shadow), calc(24% + 8% * $strength));
```
Comment (therealcoder1337) [inline src/lib/css/mixins/_design.scss:10]: I know that in math, `*` is priorizited over `+`, but let's please use braces here. Even if this is just a formatting change. ` calc(24% + (8% * $strength))`

File: src/lib/css/mixins/_design.scss [hunk 4]
```diff
@@ -59,6 +58,6 @@
   opacity: $opacity;
 }
 
-@function add-transparency($color-variable: null, $percentage: null) {
+@function with-alpha($color-variable: null, $percentage: null) {
```
Comment (therealcoder1337) [inline src/lib/css/mixins/_design.scss:61]: Better naming, good idea.

**Review by therealcoder1337 — CHANGES_REQUESTED** [review]
The changes are cool, but it seems some fine tuning is required. Please fix CI and read my comments, thanks!

File: src/lib/components/private/QDocsSection.svelte (hunk 2 above)
Comment (KilDesu) [reply src/lib/components/private/QDocsSection.svelte:21]: It has been replaced by `{#if code}`, what do you mean?

File: src/lib/css/mixins/_design.scss (hunk 3 above)
Comment (KilDesu) [reply src/lib/css/mixins/_design.scss:10]: Sure

File: package.json [hunk 5]
```diff
@@ -37,6 +37,7 @@
   "devDependencies": {
     "@fontsource/roboto": "^5.1.0",
     "@material/material-color-utilities": "^0.3.0",
+    "@shikijs/twoslash": "^1.21.0",
```
[the remark this answers is not in the document]
Comment (KilDesu) [reply package.json:40]: Not sure what you meant, but I removed the dependency anyway, I don't think we need it

File: src/lib/components/private/QDocsSection.svelte (hunk 2 above)
Comment (therealcoder1337) [reply src/lib/components/private/QDocsSection.svelte:21]: The variable `snippets` still exists in this component - why not remove it completely?

**Review by therealcoder1337 — APPROVED** [review]

**Pull request merged**